suite/tests/api/ir_riscv64.c

```diff
@@ -109,7 +109,13 @@ test_instr_encoding_jump_or_branch(void *dc, uint opcode, instr_t *instr)
     decin = instr_create(dc);
     next_pc = decode(dc, buf, decin);
     ASSERT(next_pc != NULL);
-    ASSERT(instr_same(instr, decin));
+    /* AUIPC only encodes the high 20-bit immediate value, so it is expected that the
```

Comment:
If the encoding pc remains the same and the immediate was at-most 20 bit long, I don't see why we'll get different instrs on re-encoding. Do you have an example?

Reply:
When we decode the recoded instr, we passed the `buf` as the new pc, that's how it got different. Ideally, decoding of auipc should always be given a pc that is aligned to `(1<<12)`, but I think that's not doable, right?

Reply:
For OP_auipc, we could use instr_encode_to_copy.

Reply:
Fixed in https://github.com/DynamoRIO/dynamorio/pull/6208/commits/9ab092dfefeb8675dd0323ef2d14d6cbad1d84cc